frontends/p4/functionsInlining.cpp

```diff
@@ -69,6 +69,48 @@ Visitor::profile_t FunctionsInliner::init_apply(const IR::Node *node) {
     return rv;
 }
 
+class FunctionsInliner::isLocalExpression : public Inspector, ResolutionContext {
+    bool done = false;
+    bool result = true;
+
+    profile_t init_apply(const IR::Node *node) override {
+        BUG_CHECK(!done, "isLocalExpression can only be applied once");
+        return Inspector::init_apply(node);
+    }
+    void end_apply() override { done = true; }
+    bool preorder(const IR::Node *) override { return result; }
+    bool preorder(const IR::Path *p) override {
+        if (p->absolute) return result = false;
+        // cribbed from ResolutionContext::resolve -- we want to resolve the name,
```

Comment:
Maybe we can add method to `ResolutionContext`, so it will be available if necessary elsewhere?

Reply:
I've been trying to think of ways to make this more general, but it is (currently) really specific to what is needed in this `isLocalExpression` test.  Extending the current `resolve` to also return context info with the declaration would be possible but not trivial (requires making a copy of the VisitorContext linked list if we want to save that information beyond just locally).  It would also be more expensive than this, which can short-cut a number of cases.